### PR TITLE
Fix flaky script to setup local kubernetes cluster

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR fixes the below issues in  `./kind/common.sh`
* Fixes `${PROMETHEUS_ENABLE,,}` bad substitution in case of zsh shell.
* when sed command is triggered it creates a file with <filename>.<extension>-e. When `kubectl create -f kube-prometheus/manifests/setup` is triggerd from within the script it tries to create manifests with -e filename as well which causes error in script.

CC: @SamYuan1990 @rootfs

### CODE REVIEW

**Comment by SamYuan1990** [comment]
please see CI issue. btw, what's your OS?

**Comment by vprashar2929** [comment]
> please see CI issue. btw, what's your OS?

Fixed the CI issue. Now I think it should run fine.  I ran script on MacOS(Apple Silicon)

File: kind/common.sh [hunk 1]
```diff
@@ -198,7 +204,7 @@ function _setup_kind() {
     _wait_containers_ready kube-system
     _run_registry
 
-    if [ ${PROMETHEUS_ENABLE,,} == "true" ]; then
+    if [ ${PROMETHEUS_ENABLE} == "true" ]; then
```
Comment (marceloamaral) [inline kind/common.sh:207]: same here

**Review by marceloamaral — CHANGES_REQUESTED** [review]
Looks good, I have just a minor comment

**Comment by vprashar2929** [comment]
Added review comments 🙂

**Review by SamYuan1990 — APPROVED** [review]
LGTM

File: kind/common.sh [hunk 2]
```diff
@@ -65,11 +65,17 @@ aarch64* | arm64*)
     ;;
 esac
 
+# check os
+if [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (SamYuan1990) [inline kind/common.sh:69]: I know sed on mac has different behaviors with linux. hence if the change is not impact with linux(for our CI), it looks go to me.

Comment (vprashar2929) [reply kind/common.sh:69]: Yes it won't affect other Linux distributions

**Review by SamYuan1990 — CHANGES_REQUESTED** [review]
Sorry, we have to fix #7 before merge this PR>

**Comment by SamYuan1990** [comment]
@rootfs , please help check why this PR skipped DCO check?

**Comment by rootfs** [comment]
@SamYuan1990 DCO was enabled after this PR.

**Comment by rootfs** [comment]
@vprashar2929 we are going to closing this one so the repo will pick up DCO for upcoming PRs. Please create a new PR with this work. Thanks

**Pull request closed without merging**